Dynamic file paths

IamfromSpace opened the pull request:
Fully backwards compatible and tested way to allow custom file names with dynamic timestamps/process ids.  The latter is for supporting testing that spins up child process (such as selenium grid testing).

Happy to make changes from feedback.

matthaias commented on lib/file-path-parser.js:
```diff
@@ -0,0 +1,36 @@
+// Also, testing with time/timezones is rather difficult, so that's all mocked.
+module.exports = function(myProcess, myDate, myFormatter) {
+  return function(s) {
+    r = s;
```
Can you please use better names here? It's hard for me to understand what happens. Also, i am not sure if this is needed, since `s` is not used anymore, or i didn't see it?

matthaias commented on lib/file-path-parser.js:
```diff
@@ -0,0 +1,36 @@
+    r = r.replace(simpleTSRE, myFormatter(now, 'isoDateTime'));
+
+    // collect all of our regex matches
+    var matches = [];
```
Would it be possible to move the var statements to the top of the function?

matthaias commented:
Sorry for the delayed review! I made small comments, the rest looks good!

IamfromSpace commented:
Updated!  Good call on the variables not giving good insight into what the function actually did.  Added a comment to clarify as well.  Let me know if you have any other concerns :)

matthaias commented:
Looks good, thanks! Published as `v0.0.11`.
